data/crac-creation/crac-creator-csa-profiles/src/main/java/com/farao_community/farao/data/crac_creation/creator/csa_profile/CsaProfileCrac.java

```diff
@@ -28,26 +26,50 @@ public class CsaProfileCrac implements NativeCrac {
 
     private final QueryCatalog queryCatalogCsaProfileCrac;
 
+    private Map<String, Set<String>> keywordMap;
+
     public CsaProfileCrac(TripleStore tripleStoreCsaProfileCrac) {
         this.tripleStoreCsaProfileCrac = tripleStoreCsaProfileCrac;
         this.queryCatalogCsaProfileCrac = new QueryCatalog(CsaProfileConstants.SPARQL_FILE_CSA_PROFILE);
+        this.keywordMap = new HashMap<>();
     }
 
     @Override
     public String getFormat() {
         return "CsaProfileCrac";
     }
 
+    public void clearContext(String context) {
+        tripleStoreCsaProfileCrac.clear(context);
+    }
+
+    public void fillKeywordMap(Map<String, Set<String>> keywordMap) {
+        this.keywordMap = keywordMap;
+    }
+
+    private Set<String> getContextNamesToRequest(CsaProfileKeywords csaProfileKeyword) {
+        if (keywordMap.containsKey(csaProfileKeyword.toString())) {
+            return keywordMap.get(csaProfileKeyword.toString());
+        }
+        return tripleStoreCsaProfileCrac.contextNames();
+    }
+
+    public Map<String, PropertyBags> getHeaders() {
+        Map<String, PropertyBags> returnMap = new HashMap<>();
+        tripleStoreCsaProfileCrac.contextNames().forEach(context -> returnMap.put(context, queryTripleStore(CsaProfileConstants.REQUEST_HEADER, Set.of(context))));
+        return returnMap;
+    }
+
     public PropertyBags getContingencies() {
-        return this.queryTripleStore(Arrays.asList(CsaProfileConstants.REQUEST_ORDINARY_CONTINGENCY, CsaProfileConstants.REQUEST_EXCEPTIONAL_CONTINGENCY, CsaProfileConstants.REQUEST_OUT_OF_RANGE_CONTINGENCY), tripleStoreCsaProfileCrac.contextNames());
+        return queryTripleStore(Arrays.asList(CsaProfileConstants.REQUEST_ORDINARY_CONTINGENCY, CsaProfileConstants.REQUEST_EXCEPTIONAL_CONTINGENCY, CsaProfileConstants.REQUEST_OUT_OF_RANGE_CONTINGENCY), getContextNamesToRequest(CsaProfileKeywords.CONTINGENCY));
     }
 
     public PropertyBags getContingencyEquipments() {
         return this.queryTripleStore(CsaProfileConstants.REQUEST_CONTINGENCY_EQUIPMENT);
```

Comment:
I think you should add context names associated to keyword in each request, and not only the one which have already context lists before. In that way, you'll be able to check if each data is in a valid file, and not only the main object of each file, as before.

Reply:
Done